spinner/eglspinner.cpp

```diff
@@ -317,7 +335,41 @@ try
     }
     g_clear_error(&error);
 
-    auto const surfaces = mir_eglapp_init(argc, argv);
+    std::vector<std::shared_ptr<EglSurface>> surfaces;
+    char* backendEnv = getenv("COMPOSITOR_BACKEND");
+    Backend backend = UNKOWN;
+
+    if (backendEnv == nullptr) {
+        if (session_config.is_android()) {
+            printf("Running on an Android device, using Mir output.\n");
+            backend = MIR;
+        } else {
+            printf("Not running on an Android device, using Wayland output.\n");
+            backend = WAYLAND;
+        }
+    } else {
+        printf("Using COMPOSITOR_BACKEND varable to set backend\n");
+        if (strcmp(backendEnv, "wayland") == 0) {
+            backend = WAYLAND;
+        }
+        if (strcmp(backendEnv, "mir") == 0) {
+            backend = MIR;
+        }
```

Comment:
The unknown case handling is a bit hard to follow. Maybe consider handling it right here?